tools/nl/embeddings/palm_alternatives.py

```diff
@@ -70,6 +68,31 @@ class Context:
   model: any
 
 
+def _df_svs_validated(df_svs: pd.DataFrame) -> bool:
+  # Returns true if the following validation conditions hold:
+  # 1. All rows have a DCID string.
+  # 2. All rows have a non-empty description OR
+  #    the first alternative is non-empty.
+  for rid, row in df_svs.iterrows():
+    sv_dcid = row[SHEETS_DCID_COL]
+    sv_description = row[SHEETS_DESCRIPTION_COL]
+
+    if not sv_dcid or not isinstance(sv_dcid, str):
+      print(f"Row ({rid}) in df_svs does not have a DCID.")
+      return False
+
+    if not sv_description or not isinstance(sv_description,
+                                            str) or len(sv_description) <= 3:
```

Comment:
just curious, when is sv_description not null and not str?

Reply:
Not applicable any more. Btw, this happens unless the csv -> DataFrame conversion fills "na" entries with empty strings.